### PR TITLE
Remove dependence on VA starter Gradle plugins: phase 1

### PR DESCRIPTION
**Pull request by yoomlam**
## What was the problem?

Our Gradle build configuration requires VA's starter plugins (e.g., `gov.va.starter:plugins` and `gov.va.starter:cookiecutter`). Accessing the plugins require a GitHub PAT and the GitHub user to be part of the VA org.

- #1514

## How does this fix it?

Remove dependence on VA starter Gradle plugins by pulling them into our repo.

Code comments with `TODO PR #1549` indicate follow-on actions to be addressed in another PR.

## How to test this PR

PR checks pass and end2end tests pass

1. Remove Gradle cache of va.starter plugins:
```
cd ~/.gradle/caches/modules-2/files-2.1 # may be different path for your system
rm -rf starter.* gov.va.starter
```
2. `unset GITHUB_ACCESS_TOKEN` and build locally

### CODE REVIEW

**Comment by yoomlam** [comment]
[PR: End-to-end test for v2](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4932451313)

**Comment by yoomlam** [comment]
[PR: End-to-end test for v2](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4932523111)

**Comment by yoomlam** [comment]
[PR: End-to-end test for v2](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4932800107)

**Comment by yoomlam** [comment]
[PR: End-to-end test for v2](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4932915117)

**Comment by yoomlam** [comment]
[PR: End-to-end test for v2](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4939344601)

**Comment by yoomlam** [comment]
[PR: End-to-end test for v2](https://github.com/department-of-veterans-affairs/abd-vro/actions/runs/4939830355)

File: db-init/build.gradle [hunk 1]
```diff
@@ -33,7 +34,7 @@ configurations {
 }
 
 dependencies {
-  flywayMigration platform("gov.va.starter:starter-bom:${starter_boot_version}")
+  //  flywayMigration platform("gov.va.starter:starter-bom:${starter_boot_version}")
```
Comment (yoomlam) [inline db-init/build.gradle:36]: Not sure why this was being used. It doesn't seem to be needed.

File: domain-cc/cc-api-controller/build.gradle [hunk 2]
```diff
@@ -13,7 +13,7 @@ ext {
 dependencies {
   // Used by API classes
   implementation 'io.micrometer:micrometer-core'
-  implementation "gov.va.starter:exceptions"
+  //  implementation "gov.va.starter:exceptions"
```
Comment (yoomlam) [inline domain-cc/cc-api-controller/build.gradle]: We don't use any of these exceptions: https://github.com/department-of-veterans-affairs/lighthouse-di-starter-boot/tree/release-0.1.32/exceptions/src/main/java/gov/va/starter/boot/exception

File: settings.gradle [hunk 3]
```diff
@@ -12,18 +12,21 @@ pluginManagement {
...
-      name = "starterBootPkgs"
-      url = uri("https://maven.pkg.github.com/department-of-veterans-affairs/lighthouse-di-starter-boot")
-      credentials(HttpHeaderCredentials) {
-        name = "Authorization"
-        value = "Bearer " + (System.getenv("GITHUB_ACCESS_TOKEN") ?: GITHUB_ACCESS_TOKEN)
-      }
-      authentication {
-        register("header", HttpHeaderAuthentication)
-      }
+      url "https://repo1.maven.org/maven2/"
```
Comment (yoomlam) [inline settings.gradle:16]: I had to use this alternative maven repo when `mavenCentral` was down. Adding it here so others don't waste time trying to fix it.

File: settings/checkstyle/checkstyle.xml
Comment (yoomlam) [inline settings/checkstyle/checkstyle.xml:1]: Copied file from https://github.com/department-of-veterans-affairs/lighthouse-di-starter-boot/tree/release-0.1.32/checkstyle/src/main/resources/settings/checkstyle, unmodified.

File: app/build.gradle [hunk 4]
```diff
@@ -23,6 +22,12 @@ tasks.withType(Test).named("end2endTest") {
 }
 
 dependencies {
+  // TODO PR #1549: Where are these versions declared? Maybe in spring-boot-dependencies.
```
Comment (yoomlam) [inline app/build.gradle:25]: `TODO PR #1549` indicates follow-on action in another PR.

File: app/build.gradle [hunk 5]
```diff
@@ -1,7 +1,6 @@
 plugins {
   id 'local.std.java.application-conventions'
   id 'starter.java.config-conventions'
-  id 'starter.java.build-utils-conventions'
```
Comment (yoomlam) [inline app/build.gradle:4]: This plugin isn't particularly useful and it appears in many places.

File: domain-cc/cc-api-controller/build.gradle [hunk 6]
```diff
@@ -13,7 +13,6 @@ ext {
 dependencies {
   // Used by API classes
   implementation 'io.micrometer:micrometer-core'
-  implementation "gov.va.starter:exceptions"
```
Comment (yoomlam) [inline domain-cc/cc-api-controller/build.gradle:16]: We don't use any of these exceptions: https://github.com/department-of-veterans-affairs/lighthouse-di-starter-boot/tree/release-0.1.32/exceptions/src/main/java/gov/va/starter/boot/exception

File: settings/checkstyle/checkstyle.xml
Comment (yoomlam) [inline settings/checkstyle/checkstyle.xml:1]: Copied file from https://github.com/department-of-veterans-affairs/lighthouse-di-starter-boot/tree/release-0.1.32/checkstyle/src/main/resources/settings/checkstyle, unmodified.

**Review by msnwatson — APPROVED** [review]

**Comment by yoomlam** [comment]
red nodes = plugin directly referenced by VRO projects
image

**Pull request merged**